packages/grafana-ui/src/components/Slider/Slider.test.tsx

```diff
@@ -10,46 +12,73 @@ const sliderProps: SliderProps = {
-    expect(wrapper.html()).toContain('aria-valuemax="20"');
-    expect(wrapper.html()).toContain('aria-valuenow="10"');
+    render(<Slider {...sliderProps} />);
+
+    const slider = screen.getByRole('slider');
+    const sliderInput = screen.getByRole('textbox');
+
+    expect(slider).toHaveAttribute('aria-valuemin', '10');
+    expect(slider).toHaveAttribute('aria-valuemax', '20');
+    expect(slider).toHaveAttribute('aria-valuenow', '10');
+    expect(sliderInput).toHaveValue('10');
   });
 
   it('renders correct contents with a value', () => {
-    const wrapper = mount(<Slider {...sliderProps} value={15} />);
-    expect(wrapper.html()).toContain('aria-valuenow="15"');
-    expect(wrapper.html()).not.toContain('aria-valuenow="20"');
-    expect(wrapper.html()).not.toContain('aria-valuenow="10"');
+    render(<Slider {...sliderProps} value={15} />);
+
+    const slider = screen.getByRole('slider');
+    const sliderInput = screen.getByRole('textbox');
+
+    expect(slider).toHaveAttribute('aria-valuenow', '15');
+    expect(sliderInput).toHaveValue('15');
   });
 
-  it('allows for custom values to be set in the input', () => {
-    const wrapper = mount(<Slider {...sliderProps} value={10} min={10} max={100} />);
-    const sliderInput = wrapper.find('input');
-    sliderInput.simulate('focus');
-    sliderInput.simulate('change', { target: { value: 50 } });
-    sliderInput.simulate('blur');
-    expect(wrapper.html()).toContain('aria-valuenow="50"');
+  it('allows for custom values to be set in the input', async () => {
+    render(<Slider {...sliderProps} value={10} min={10} max={100} />);
+
+    const slider = screen.getByRole('slider');
+    const sliderInput = screen.getByRole('textbox');
+
+    await user.clear(sliderInput);
+    await user.type(sliderInput, '50');
+
+    expect(slider).toHaveAttribute('aria-valuenow', '50');
+    expect(sliderInput).toHaveValue('50');
+
+    // click outside the input field to blur
+    await user.click(document.body);
+
+    expect(slider).toHaveAttribute('aria-valuenow', '50');
+    expect(sliderInput).toHaveValue('50');
   });
 
-  it('defaults after blur if input value is outside of range', () => {
-    const wrapper = mount(<Slider {...sliderProps} value={10} min={10} max={100} />);
-    const sliderInput = wrapper.find('input');
-    sliderInput.simulate('focus');
-    sliderInput.simulate('change', { target: { value: 200 } });
-    // re-grab to check value is out of range before blur
-    const sliderInputIncorrect = wrapper.find('input');
-    expect(sliderInputIncorrect.get(0).props.value).toEqual('200');
-
-    sliderInput.simulate('blur');
-    expect(wrapper.html()).toContain('aria-valuenow="100"');
-    // re-grab to check value is back inside range
-    const sliderInputCorrect = wrapper.find('input');
-    expect(sliderInputCorrect.get(0).props.value).toEqual('100');
+  it('sets value back to default after blur if input value is outside of range', async () => {
```

Comment:
**Nit:** maybe something like this would cover better what it is doing? I would probably also extend it with a few lines to check if it works for the minimum value as well.

```suggestion
  it('sets value to the closest available one after blur if input value is outside of range', async () => {
```